### PR TITLE
Support enum init fields in gapic config

### PR DESCRIPTION
**Pull request by shinfan**
- Example config:
`book.rating=GOOD`
- Also adds missing import "java.util.Arrays" into generated smoke test.

Fixes https://github.com/googleapis/toolkit/issues/733

### CODE REVIEW

**Comment by shinfan** [comment]
Note: The travis fail should be irrelevant to this PR.

File: src/main/java/com/google/api/codegen/metacode/InitCodeNode.java [hunk 1]
```diff
@@ -462,45 +463,53 @@ private static FieldConfig getChildFieldConfig(
    * if the provided type is not supported or doesn't match the value.
    */
   private static void validateValue(TypeRef type, String value) {
-    Type descType = type.getKind();
-    switch (descType) {
-      case TYPE_BOOL:
-        String lowerCaseValue = value.toLowerCase();
-        if (lowerCaseValue.equals("true") || lowerCaseValue.equals("false")) {
+    if (type.isEnum()) {
```
Comment (michaelbausor) [inline src/main/java/com/google/api/codegen/metacode/InitCodeNode.java]: You can use the switch statement and check for TYPE_ENUM

Comment (shinfan) [inline src/main/java/com/google/api/codegen/metacode/InitCodeNode.java]: Will do.

**Comment by shinfan** [comment]
@michaelbausor  PTAL

**Comment by michaelbausor** [comment]
LGTM

**Pull request merged**